include/closure_hydrator.h

```diff
@@ -62,6 +62,8 @@ class closure_hydrator {
 		range<3> backing_buffer_range;
 		id<3> backing_buffer_offset;
 		subrange<3> accessed_virtual_subrange;
+		// Only used when CELERITY_ACCESSOR_BOUNDARY_CHECK=1
+		id<3>* out_of_bounds_indices = nullptr;
```

Comment:
Why is this not just `#if`'d`?

Reply:
discussed offline, we keep it here for uniform handling in tests

Reply:
This is based on an earlier suggestion of mine, since I thought having this included conditionally meant that we'd have to also have to branch every time we create a `closure_hydrator::accessor_info`. However since aggregate initialization allows to only partially initialize PODs we actually don't, so it would be fine to bring back the `#if`.